src/storage/mvcc/txn.rs

```diff
@@ -203,9 +219,92 @@ impl<S: Snapshot> MvccTxn<S> {
         Ok(())
     }
 
+    pub fn pessimistic_lock(
+        &mut self,
+        key: Key,
+        primary: &[u8],
+        for_update_ts: u64,
+        options: &Options,
+    ) -> Result<()> {
+        if let Some(lock) = self.reader.load_lock(&key)? {
+            if lock.ts != self.start_ts {
+                return Err(Error::KeyIsLocked {
+                    key: key.to_raw()?,
+                    primary: lock.primary,
+                    ts: lock.ts,
+                    ttl: lock.ttl,
+                });
+            }
+            MVCC_DUPLICATE_CMD_COUNTER_VEC.pessimistic_lock.inc();
+            return Ok(());
+        }
+
+        if let Some((commit, write)) = self.reader.seek_write(&key, u64::max_value())? {
+            // The isolation level of pessimistic transactions is RC. `for_update_ts` is
```

Comment:
Why not use `read_ts` instead of `for_update_ts`?

Reply:
`for_update_ts` is what used in unistore. And I think `lock_ts` might be more proper here.

Reply:
This name is originally used by `select for update`, then later extended to be used by `update/delete/insert`, it is used in many places. I think we better keep it.